core/trino-main/src/main/java/io/trino/server/Server.java

```diff
@@ -142,6 +143,11 @@ private void doStart(String trinoVersion)
 
             injector.getInstance(PluginInstaller.class).loadPlugins();
 
+            var catalogStoreManager = injector.getInstance(new Key<Optional<CatalogStoreManager>>() {});
```

Comment:
We have a utility method in this class:
```java
var catalogStoreManager = injector.getInstance(optionalKey(CatalogStoreManager.class));
```
Though this is cool -- I didn't know you could use `Key` directly like this. I've only seen that done with `TypeLiteral`. We could cleanup lots of code with this. Apparently this has always been supported since the first version of Guice.